### ISSUE TITLE
Dockerfile does not build (thirdeye-ui)

### ISSUE THREAD

**Issue by tfrokt** [startreedata/thirdeye#1245]
I am trying to get thirdeye set up locally to play around with it. I've managed to solve a few issues along the way but I am stuck now at building the ui.

Steps to reproduce:
```
cd thirdeye-ui
docker build -t thirdeye-ui .
```

Error:
```
[..]
ERROR [6/8] COPY dist/ /etc/nginx/html
[..]
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref f8b9e55d-88d3-4e91-9aad-64baa8cd33aa::ktlorb2wargzic7x4g94tnaac: "/dist": not found
```

I cannot find any "dist" folder in the repository.

**Comment by cyrilou242** [comment]
Hey @tfrokt, 
The dockerfile assumes the UI is built outside of the dockerfile. 

You should be able to build the docker image doing: 
```
cd thirdeye-ui
npm run build
docker build -t thirdeye-ui .
```

**Comment by cyrilou242** [comment]
If you don't want to install npm and other tools required to install the UI, we are considering building the UI inside the docker image. 
We had a PR to build the UI in the dockerfile directly 
https://github.com/startreedata/thirdeye/pull/1168
but it's not ready yet.

**Comment by tfrokt** [comment]
Salut @cyrilou242 ,

Thanks for the quick reply! After trying getting it to work outside of Docker (using the latest npm) I finally managed to get it to run based off the PR you linked. I had to adjust a few things - and it is not pretty, but it should be enough for me to play around with it. 

Have attached the diff for the record, but it would definitely require more work to be cleaned up. 
[docker-compose.patch](https://github.com/startreedata/thirdeye/files/13552372/docker-compose.patch)
